### PR TITLE
Support sample.env generation in rlc-common for typespec-ts

### PR DESCRIPTION
**Pull request by kazrael2119**
fixes https://github.com/Azure/autorest.typescript/issues/2806

### CODE REVIEW

File: packages/rlc-common/src/metadata/buildSampleEnvFile.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+import { RLCModel } from "../interfaces.js";
+
+const sampleEnvText = `
+# App registration secret for AAD authentication
+AZURE_CLIENT_SECRET=
```
Comment (MaryGao) [inline packages/rlc-common/src/metadata/buildSampleEnvFile.ts:7]: Could you check if this is still valid?

File: packages/rlc-common/src/metadata/buildSampleEnvFile.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+AZURE_CLIENT_ID=
+AZURE_TENANT_ID= 
+`;
+
+export function buildSampleEnvFile(model: RLCModel) {
+  if (
+    model.options?.generateMetadata === true ||
+    model.options?.generateSample === true
```
Comment (MaryGao) [inline packages/rlc-common/src/metadata/buildSampleEnvFile.ts:15]: add one condition: only generated this file in azure packages

File: packages/rlc-common/src/metadata/buildSampleEnvFile.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import { RLCModel } from "../interfaces.js";
+
+const sampleEnvText = `
+# App registration secret for AAD authentication
+AZURE_CLIENT_SECRET=
+AZURE_CLIENT_ID=
+AZURE_TENANT_ID= 
+`;
+
+export function buildSampleEnvFile(model: RLCModel) {
```
Comment (MaryGao) [inline packages/rlc-common/src/metadata/buildSampleEnvFile.ts:12]: Could you share the failure message with me if no sample.env provided?

Comment (kazrael2119) [reply packages/rlc-common/src/metadata/buildSampleEnvFile.ts:12]: image

**Review by MaryGao — CHANGES_REQUESTED** [review]
Need to verify if the env var works!

File: packages/rlc-common/src/metadata/buildSampleEnvFile.ts (hunk 1 above)
Comment (qiaozha) [reply packages/rlc-common/src/metadata/buildSampleEnvFile.ts:7]: @HarshaNalluru could you hep check if tthis is expected as of the security concern of spring groove

**Pull request closed without merging**